### PR TITLE
feat: general swap

### PR DESCRIPTION
**Pull request by haythem96**
# feat: general swap

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

**Review by alpinechicken — APPROVED** [review]
lgtm

File: packages/hardhat/contracts/periphery/ControllerHelper.sol [hunk 1]
```diff
@@ -786,6 +804,11 @@ contract ControllerHelper is UniswapControllerHelper, AaveControllerHelper, IERC
                 ControllerHelperDiamondStorage.getAddressAtSlot(3),
                 _amountToPay
             );
+        } else if (
+            ControllerHelperDataType.CALLBACK_SOURCE(_callSource) ==
+            ControllerHelperDataType.CALLBACK_SOURCE.GENERAL_SWAP
+        ) {
+            IERC20(_tokenIn).transfer(ControllerHelperDiamondStorage.getAddressAtSlot(3), _amountToPay);
```
Comment (aleone) [inline packages/hardhat/contracts/periphery/ControllerHelper.sol:811]: this only works for WETH/ETH to oSQTH swaps when we have a WETH balance in the contract

any payable ETH can't be used for this as it isn't ever wrapped and any withdrawn ETH from a vault can't be used for it as it is withdrawn as ETH.

only WETH from an existing uniswap position could be used for this.  One potential solution is wrapping ETH balance before the callback and unwrapping weth after transfering the amount owed from the flashswap.

Comment (aleone) [reply packages/hardhat/contracts/periphery/ControllerHelper.sol:811]: Another potential solution is to send some ETH to uniswap as part of the call or repay with ETH, this possible?

**Review by aleone — CHANGES_REQUESTED** [review]
See comments on wrapping weth, swaps are limited from ETH to oSQTH as we don't wrap any withdrawn ETH or payable ETH to weth.

File: packages/hardhat/contracts/periphery/ControllerHelper.sol (hunk 1 above)
Comment (haythem96) [reply packages/hardhat/contracts/periphery/ControllerHelper.sol:812]: I will just wrap the contract balance again, also to be consistent by repaying WETH to pool (not sending ETH)

**Review by aleone — APPROVED** [review]
lgtm

**Pull request merged**